hedera-node/hedera-app-spi/src/test/java/com/hedera/node/app/spi/config/internal/KeyValuePairConverterTest.java

```diff
@@ -0,0 +1,133 @@
+
+        // then
+        assertThat(result.key()).isEqualTo("key");
+        assertThat(result.value()).isEqualTo("value");
+    }
+
+    @Test
+    void testValidValueMultipleSemicolon() {
+        // given
+        final KeyValuePairConverter converter = new KeyValuePairConverter();
+        final String value = "key;value;stillValue;";
+
+        // when
+        final var result = converter.convert(value);
+
+        // then
+        assertThat(result.key()).isEqualTo("key");
+        assertThat(result.value()).isEqualTo("value;stillValue;");
+    }
+
+    @Test
+    void testInvalidValueNoSemicolon() {
+        // given
+        final KeyValuePairConverter converter = new KeyValuePairConverter();
+        final String value = "key";
+
+        // then
+        assertThatThrownBy(() -> converter.convert(value)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void testIncludedInConfig() {
+        // given
+        final Configuration configuration = ConfigurationBuilder.create()
+                .withConverter(new KeyValuePairConverter())
+                .withSource(new SimpleConfigSource("key", "key;value"))
+                .build();
+
+        // when
+        final KeyValuePair value = configuration.getValue("key", KeyValuePair.class);
+
+        // then
+        assertThat(value).isNotNull();
+        assertThat(value.key()).isEqualTo("key");
+        assertThat(value.value()).isEqualTo("value");
+    }
+
+    @Test
+    void testIncludedInListInConfig() {
+        // given
+        final Configuration configuration = ConfigurationBuilder.create()
+                .withConverter(new KeyValuePairConverter())
+                .withSource(new SimpleConfigSource("key", "key;value"))
+                .build();
+
+        // when
+        final List<KeyValuePair> values = configuration.getValues("key", KeyValuePair.class);
+
+        // then
+        assertThat(values).isNotNull().hasSize(1);
+        assertThat(values.get(0).key()).isEqualTo("key");
+        assertThat(values.get(0).value()).isEqualTo("value");
+    }
+
+    @Test
+    void testMultipleIncludedInListInConfig() {
+        // given
+        final Configuration configuration = ConfigurationBuilder.create()
+                .withConverter(new KeyValuePairConverter())
+                .withSource(new SimpleConfigSource("key", "key1;value1,key2;value2,key3;value3"))
+                .build();
+
+        // when
+        final List<KeyValuePair> values = configuration.getValues("key", KeyValuePair.class);
+
+        // then
+        assertThat(values).isNotNull().hasSize(3);
+        assertThat(values.get(0).key()).isEqualTo("key1");
+        assertThat(values.get(0).value()).isEqualTo("value1");
+        assertThat(values.get(1).key()).isEqualTo("key2");
+        assertThat(values.get(1).value()).isEqualTo("value2");
+        assertThat(values.get(2).key()).isEqualTo("key3");
+        assertThat(values.get(2).value()).isEqualTo("value3");
+    }
+}
```

Comment:
I would also add tests for `Set`, just in case.

Reply:
That is not possible since the support for Set in the config is not merged yet ;)

Reply:
Ah, right. Forgot that.